app/controllers/admin/users_controller.rb

```diff
@@ -13,6 +13,25 @@ def index
       end
     end
 
+    def new
+      @user = User.new
+    end
+
+    def create
+      user_params = params.require(:user)
+      user_type = user_params[:type].camelize.downcase
+      case user_type
```

Comment:
Unsure if this is the best way to check if the type is valid

Reply:
The `case` is okay, but don't do the `camelize` and `downcase` stuff, just ensure that the input values are the literal names of the classes (`AdminUser`/`EndUser`).